### ISSUE TITLE
Localise the flip/force a flip

### ISSUE THREAD

**Issue by wroughtec** [jshjohnson/Choices#116]
I have been experimenting switching from Select2 to Choices as really like how clean Choices is.

I have a dropdown which is fixed at the bottom of the screen (the html is much higher up) and with choices currently its trying to open the dropdown below when obviously there is no room to open the dropdown, in select2 this works.

I think its in the way it calculates of the viewport size:
```javascript
          var viewport = {
            top: $window.scrollTop(),
            bottom: $window.scrollTop() + $window.height()
          };

          var enoughRoomAbove = viewport.top < (offset.top - dropdown.height);
          var enoughRoomBelow = viewport.bottom > (offset.bottom + dropdown.height);

          var css = {
            left: offset.left,
            top: container.bottom
          };

          // Determine what the parent element is to use for calciulating the offset
          var $offsetParent = this.$dropdownParent;

          // For statically positoned elements, we need to get the element
          // that is determining the offset
          if ($offsetParent.css('position') === 'static') {
            $offsetParent = $offsetParent.offsetParent();
          }

          var parentOffset = $offsetParent.offset();

          css.top -= parentOffset.top;
          css.left -= parentOffset.left;

          if (!isCurrentlyAbove && !isCurrentlyBelow) {
            newDirection = 'below';
          }

          if (!enoughRoomBelow && enoughRoomAbove && !isCurrentlyAbove) {
            newDirection = 'above';
          } else if (!enoughRoomAbove && enoughRoomBelow && isCurrentlyAbove) {
            newDirection = 'below';
          }

          if (newDirection == 'above' ||
            (isCurrentlyAbove && newDirection !== 'below')) {
            css.top = container.top - parentOffset.top - dropdown.height;
          }
```

So it takes into account if the dropdown would be visible inside the viewport where as (and forgive me if i am wrong I only briefly looked into the code) Choices seems to calculate the entire window size and as long as the dropdown would fit (regardless if you have to scroll down) then it will open below. Is this something that could be possible or perhaps an easier solution would a config to force the dropdown to go above.

**Comment by wroughtec** [comment]
After a quick experiment
```javascript
// Changing:
Math.max(body.scrollHeight, body.offsetHeight, html.clientHeight, html.scrollHeight, html.offsetHeight);

// to:
Math.min(body.scrollHeight, body.offsetHeight, html.clientHeight, html.scrollHeight, html.offsetHeight);
```

Seems to have resolved my issue but not 100% sure this would be a safe solution for everyone

**Comment by jshjohnson** [comment]
Hey,

Thanks for looking into this! I'm guessing for your situation it would make more sense to enforce the drop down above or below instead of detecting it using the space available? As the mechanism to flip the drop down is already there, that would be a pretty trivial thing to implement. 

Let me know what you think!

**Comment by wroughtec** [comment]
Yeah that would help thanks

**Comment by jshjohnson** [comment]
I have just implemented this in version [2.7.4](https://github.com/jshjohnson/Choices/releases/tag/v2.7.4).

See the new `position` config option [here](https://github.com/jshjohnson/Choices#position).
